aodncore/pipeline/files.py

```diff
@@ -788,7 +806,32 @@ def set_file_update_callback(self, file_update_callback):
         for f in self.__s:
             f.file_update_callback = file_update_callback
 
-    def set_default_publish_types(self, include_regexes, exclude_regexes, addition_type, deletion_type):
+    def set_check_types_from_params(self, check_params):
+        """Set check_type attribute for each file in the collection to the default value, based on the file type and
+        presence of compliance checker checks in the check parameters
+
+        :param check_params: :py:class:`dict`
+        :return:
+        """
+        if check_params is None:
+            check_params = {}
+        else:
+            validate_check_params(check_params)
+
+        checks = check_params.get('checks', ())
+
+        all_deletions = PipelineFileCollection(f for f in self.__s if f.is_deletion)
+        all_additions = PipelineFileCollection(self.__s.difference(all_deletions))
```

Comment:
Could this just be `self.difference(all_deletions)`?

Reply:
It probably *could* be but the reasoning is that I've been trying to avoid public methods which loop back and call other public methods on the same instance. Everything else operates against the internal set, so I didn't want to change this convention just for this one.

It could potentially be slackened off overall to use the public methods from the same instance(for example, to reduce duplication of similar operations against the set in multiple places), but I'd want that to be a conscious choice for everything at once I think (and in a different PR).